tools/functional-tester/etcd-runner/command/lock_racer_command.go

```diff
@@ -69,12 +72,16 @@ func runRacerFunc(cmd *cobra.Command, args []string) {
 		m := concurrency.NewMutex(s, racers)
 		rcs[i].acquire = func() error { return m.Lock(ctx) }
 		rcs[i].validate = func() error {
+			mu.Lock()
```

Comment:
`The follower acquires the election but the doRounds code is assuming exclusive ownership and that's causing a deadlock.`

Follower can't win the election if it is a follower. Only one client can win the election. The deadlock is caused when the follower block on [`rcNextc`](https://github.com/coreos/etcd/blob/master/tools/functional-tester/etcd-runner/command/election_command.go#L135) when it obtained [`mu.Lock()`](https://github.com/coreos/etcd/blob/master/tools/functional-tester/etcd-runner/command/global.go#L86) before leader does.  However, if leader obtains  [`mu.Lock()`](https://github.com/coreos/etcd/blob/master/tools/functional-tester/etcd-runner/command/global.go#L86) before follower does, then everything is fine.

remove the lock fixes the above issue.